### ISSUE TITLE
oAuth Setup does not seem to be working

### ISSUE THREAD

**Issue by t1mb0b** [RaynorUE/snich#213]
**Describe the bug**
I have a number of instances configured, and have been using the plugin to edit and update code for some time now. Recently, the connections to instances stopped working, and I am seeing a popup stating:

SNICH record pull resulted in error running the contributed command 'snich.instance.pull_record' failed

So, I thought I would just re-configure the instance, as I have had issues in the past and when I did that, it started working again

However, once I get to the last step, with the oAuth callback, I am not getting anything on the URL which allows me to click the 'allow' button. I've tried on multiple browsers on multiple machines, and am getting the blank page and cannot click the 'allow' button to complete the setup

**To Reproduce**
1. Access the 'Setup new instance' command
2. Populate the information for the 'oAuth' setup (client secret, client id, etc)
3. Once browser window opens to complete the setup, there is no 'allow' button, just a blank page

**Expected behavior**
At the last part of the setup, I should see an 'allow' button on the webpage which opens so I can contimue and complete the oAuth setup

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10 and Linux, using Opera, Edge, and Firefox browsers

**Comment by RaynorUE** [comment]
Thanks for all this detail!

Can you confirm if you're using [SNICH](https://marketplace.visualstudio.com/items?itemName=integrateNate.snich) or [SNICH Canary](https://marketplace.visualstudio.com/items?itemName=integrateNate.snich-canary)?

I have had some lingering updates I need to merge into the core app from SNICH-Canary that improves a lot of these flows. but want to make sure i'm tracking down in the right version..

If you're not on SNICH Canary can you switch to that and give it a shot? Weirdly it's the most stable of the two items out there right now (I Just need to publish current Canary build to SNICH Prime... just a huge release notes list to build for that.. :( )

**Comment by t1mb0b** [comment]
Thanks Raynor! I am using Canary!

**Comment by RaynorUE** [comment]
Ahh, thanks for confirming. Oddly I think I was noticing that the other day
as well! I'll get to the bottom of it this weekend. If not sooner!

**Comment by t1mb0b** [comment]
One other thing I noticed, not sure if it's an issue- the Logo URL configured in the oAuth profile does not exist, it returns a 404:

https://github.com/RaynorUE/snich/blob/master/images/icon-sn-oauth.PNG%3Fraw=true

**Comment by RaynorUE** [comment]
Hrrm yea that is definitely odd... Uhm, I wouldn't THINK the image is
affecting it, but it could? You could clear that value on the oAuth profile
without issue and see if that's it.

If you're on your corporate network / VPN It could be doing something that
wouldn't throw errors if it's intercepting the page or URL (I.e. detecting
the text localhost in there and doing something weird). If you inspect the
dom / open the console and look at the dom (Elements tab/section) is there
anything there that makes it look like the page is rendering and being
hidden? Or is it completely empty? If it's also completely empty, then yea
something is gobbling up that HTML content as your computer is "downloading
it".

The only other thing that's weird or at least makes me think it's nothing
specifically coded in SNICH is that your still at the SN URL, which is all
on SN for how that page renders / behaves. All I control is the "URL" you
go to (when VSCode asks you to open your browser) and the query parameters
on it. Which SN gives errors on the page if you get them wrong.

**Comment by t1mb0b** [comment]
I just discovered if I delete the redirect_url parameter and then try to load the page again, it now displays the page with the allow button. Not sure why having the localhost in the redirect URL is an issue. I'll have to try this on a different network and confirm if it is in fact something on my home network.

**Comment by t1mb0b** [comment]
The issue with the oAuth setup appears to be something on my local network; removing the redirect_url parameter allows for the setup to complete. 

The other issue I was seeing with the pull error may have been due to an expired oAuth token. I saw some messages in the debug logs when I turned that on. I removed the 'snich.x' files in the .vscode folder and re-ran the authentication setup (removing the redirect_url parameter in the callback), and everything seems to be working fine.

So, it appears it was all issues on my end. Thanks for taking the time to assist with this! I learned a few things. I really like the plugin! Looking forward to upcoming enhancements!
